### ISSUE TITLE
"param ?callable $callback" misinterpreted as undocumented parameter

### ISSUE THREAD

**Issue by Krinkle** [doxygen/doxygen#10344]
**Describe the bug**
A parameter that is documented is misinterpreted as undocumented parameter, including a false warning, and an HTML rendering that exposes the internal interpretation of the type being a param name and the name being the description.

**Screenshots**

https://doc.wikimedia.org/mediawiki-core/master/php/classMWDebug.html#a3279b20374f1f7367f0d02755b38f36a

<img width="608" alt="Screenshot" src="https://github.com/doxygen/doxygen/assets/156867/9771fbb5-7d10-4b15-81cc-4f359b3424f3">

```
/Users/krinkle/Development/mediawiki/includes/debug/MWDebug.php:405: warning: The following parameter of MWDebug\filterDeprecationForTest(string $regex, ?callable $callback=null) is not documented:
  parameter '$callback'
```

**To Reproduce**
```php
class MWDebug {
	/**
	 * Deprecation messages matching the supplied regex will be suppressed.
	 * Use this to filter deprecation warnings when testing deprecated code.
	 *
	 * @param string $regex
	 * @param ?callable $callback To call if $regex is hit
	 */
	public static function filterDeprecationForTest(
		string $regex, ?callable $callback = null
	): void {
		if ( !defined( 'MW_PHPUNIT_TEST' ) && !defined( 'MW_PARSER_TEST' ) ) {
			throw new LogicException( __METHOD__ . ' can only be used in tests' );
		}
		self::$deprecationFilters[$regex] = $callback;
	}
}
```

**Version**
Reproduced on Doxygen 1.9.1 (Debian Linux) and Doxygen 1.9.8 (macOS/Homebrew).

**Additional context**
The HTML rendering shows that `?callable` was parsed correctly as the type when rendering the signature, but then further down in the parameter docs, it is somehow seeing a token that lacks a dollar sign as being the parameter name. I'm guessing something somewhere isn't yet aware of nullable types.

https://www.php.net/manual/en/language.types.declarations.php

**Comment by albert-github** [comment]
Quick analysis gives that the `?` in front of the types (i.e. saying for the type: "A single base type declaration can be marked nullable by prefixing the type with a question mark (`?`)" ins not handled by the `\param` command.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10347

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
